lib/src/util/csv_parser.dart

```diff
@@ -0,0 +1,78 @@
+/// CSVParser parses a stream of lines into a list of vals compliant with
+/// the CSV standard
+///
+/// See RFC4180 for details on CSV standard
+class CsvParser {
+
+  /// Takes a single line and parses it into a list of values according to the
+  /// csv standard (see RFC4180)
+  static List<String> parseLine(
+      String line) {
+    final records = <String>[];
+    var i = 0;
+    StringBuffer record;
+    while (i < line.length) {
```

Comment:
These parsing functions take a start index, return a final index, and then write to a caller provided string buffer.

The index paradigm adds a lot of verbosity and potential bugs to the code-ideally we'd use a single iterator instance shared across all the parsing functions so that, by calling next, each parsing function advances the iterator by as many characters as it consumes. However, Dart's core libraries are bad and should feel bad and don't have built in `iterator()` methods.

May be worth creating a mixin for the String class to add a StringIterator.

Reply:
> May be worth creating a mixin for the String class to add an iterator method

If you feel like doing it go ahead

Reply:
Refactor to support newline escaping made this a moot point, closing.